Fix docs: missing parameter description

macartur opened the pull request.

macartur committed:
Improving kytos documentation

- Fixing documentation links
- Adding parameters information for some methods
- Ignoring D301 to accept to use '\' without 'r"""'

macartur committed:
Changing Parameters to Args.

diraol commented:
@macartur Please, use a better PR title and description.

And do not forget to do the PR to both `master` and `2017.1rc` branches.

One more thing, why ignoring `D301`?

cemsbr reviewed (changes requested):
I wouldn't ignore D301, but use r""" where it happens. It is OK to rebase in this specific case. Please, write a comment like "Close #441" in a commit long description.

diraol commented on kytos/core/buffers.py:
```diff
@@ -4,7 +4,7 @@
 from kytos.core.events import KytosEvent
 
-__all__ = ('KytosBuffers', )
+__all__ = ('KytosBuffers', 'KytosEventBuffer')
```
`KytosEventBuffer` was not supposed to be exposed.

diraol commented on kytos/core/buffers.py:
```diff
@@ -29,6 +29,10 @@ def put(self, event):
         received.
+
+        Args:
+            event (:class:`~kytos.core.buffers.KytosEventBuffer`):
```
event is an instance of `KytosEvent`, a class from `kytos.core.events` module.

diraol commented on kytos/core/buffers.py:
```diff
@@ -40,7 +44,12 @@ def put(self, event):
+        """Remove and return a event from top of queue.
+
+        Returns:
+            :class:`~kytos.core.buffers.KytosEventBuffer`:
```
This is not the correct class.
`kytos.core.buffers.KytosEventBuffer` != `kytos.core.events.KytosEvent`

diraol commented on kytos/core/buffers.py:
```diff
@@ -29,6 +29,10 @@ def put(self, event):
+
+        Args:
+            event (:class:`~kytos.core.buffers.KytosEventBuffer`):
+                KytosEventBuffer sent to queue.
```
Wrong here too.

diraol commented on kytos/core/buffers.py:
```diff
@@ -83,18 +92,25 @@ class KytosBuffers(object):
 
     def send_stop_signal(self):
-        """Method usd to send a kytos/core.shutdown event to all buffers."""
+        """Method to send a ``kytos/core.shutdown`` event to all buffers."""
```
Since we are rewriting, let's go with:
`Send a ``kytos/core.shutdown`` event to each buffer.`

diraol commented on kytos/core/switch.py:
```diff
@@ -373,6 +391,12 @@ def last_flood(self, ethernet_frame):
+           ethernet_frame (:class:`pyof.foundation.basic_types.Ethernet`):
+               Ethernet instance to be verified.
+        Returns:
+           datetime.datetime.now: Last time when the ethernetframe was flooded.
```
`s/ethernetframe/ethernet_frame/`

diraol commented on kytos/core/tcp_server.py:
```diff
@@ -30,10 +30,10 @@ def __init__(self, server_address, RequestHandlerClass, controller):
-            controller (KytosController): The controller instance.
-
+            controller (:class:`~kytos.core.controller.Controller`):
+                The controller instance.
```
`s/The controller instance/An instance of Kytos Controller class/`

macartur committed:
Documentation: small fixes.

- Removing KytosEventBuffer from __all__
- Changing references from KytosEventBuffer to KytosEvent

macartur committed:
Fixing python-openflow links

- Adding external reference to http://docs.kytos.io/python-openflow
- Removing ignore D301 from setup.cfg
- Close #441

diraol reviewed (approved):
I think that this is the best solution for now, despite some disadvantages. As we intend to remove all python-openflow dependencies from kytos, I recommend this PR to be accepted to solve some documentation/linter warnings and errors.
